### PR TITLE
test: add adapter tests

### PR DESCRIPTION
**Pull request by vr-varad**

### CODE REVIEW

**Review by twlite — CHANGES_REQUESTED** [review]
can you test these via `yasumu.{mock}` api instead? For example, instead of instantiating `ApplicationMock`, you could do `yasumu.app.getName()`

File: packages/testing/__test__/application.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { beforeAll, describe, expect, test } from 'vitest';
+import { ApplicationMock, yasumu } from '../src';
+
+describe('Application Mock', () => {
+  let app: ApplicationMock;
```
Comment (twlite) [inline packages/testing/__test__/application.spec.ts:5]:
```suggestion
  const app = yasumu.app;
```

File: packages/testing/__test__/application.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  
+  beforeAll(() => {
```
Comment (twlite) [inline packages/testing/__test__/application.spec.ts:7]:
```suggestion
```

File: packages/testing/__test__/application.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    app = yasumu.app;
```
Comment (twlite) [inline packages/testing/__test__/application.spec.ts:8]:
```suggestion
```

File: packages/testing/__test__/application.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  });
```
Comment (twlite) [inline packages/testing/__test__/application.spec.ts:9]:
```suggestion
```

File: packages/testing/__test__/application.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { beforeAll, describe, expect, test } from 'vitest';
+import { ApplicationMock, yasumu } from '../src';
+
+describe('Application Mock', () => {
+  let app: ApplicationMock;
+
```
Comment (twlite) [inline packages/testing/__test__/application.spec.ts:6]:
```suggestion
```

Comment (twlite) [inline packages/testing/__test__/application.spec.ts:6]:
```suggestion
```

**Review by twlite — CHANGES_REQUESTED** [review]
beforeAll hook is not needed for this, it is only useful in scenarios like

```ts
let workspace: Workspace;

beforeAll(async () => {
  workspace = await yasumu.openWorkspace({path})
})
```

**Comment by vr-varad** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/testing/__test__/command.spec.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    expect(listener.plugin).toBe('plugin');
+  });
+
+  test('should invoke command', async () => {
+    const result = await command.invoke('ping');
+    expect(result).toBe('Pong!');
+  });
+
+  test('should throw error to  handle non-existent command ', async () => {
+    try {
```
Comment (twlite) [inline packages/testing/__test__/command.spec.ts:24]:
```suggestion
    expect(yasumu.command.invoke('unknown')).rejects.toThrowError('Command unknown not found');
```

File: packages/testing/__test__/command.spec.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      await command.invoke('unknown');
```
Comment (twlite) [inline packages/testing/__test__/command.spec.ts:25]:
```suggestion
```

File: packages/testing/__test__/command.spec.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    } catch (error) {
```
Comment (twlite) [inline packages/testing/__test__/command.spec.ts:26]:
```suggestion
```

File: packages/testing/__test__/command.spec.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+      expect(error).toBeDefined();
```
Comment (twlite) [inline packages/testing/__test__/command.spec.ts:27]:
```suggestion
```

File: packages/testing/__test__/command.spec.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+      expect(error.message).toMatch('Command unknown not found'); // Adjust error handling based on implementation
```
Comment (twlite) [inline packages/testing/__test__/command.spec.ts:28]:
```suggestion
```

File: packages/testing/__test__/command.spec.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    }
```
Comment (twlite) [inline packages/testing/__test__/command.spec.ts:29]:
```suggestion
```

**Review by twlite — CHANGES_REQUESTED** [review]

**Review by twlite — CHANGES_REQUESTED** [review]

**Review by twlite — APPROVED** [review]

**Pull request merged**